contracts/script/deploy/mainnet/DeployMainnet.s.sol

```diff
@@ -0,0 +1,307 @@
+import {EjectionManager} from "lib/eigenlayer-middleware/src/EjectionManager.sol";
+import {IServiceManager} from "lib/eigenlayer-middleware/src/interfaces/IServiceManager.sol";
+import {EigenDATypesV2 as DATypesV2} from "src/core/libraries/v2/EigenDATypesV2.sol";
+import {OperatorStateRetriever} from "lib/eigenlayer-middleware/src/OperatorStateRetriever.sol";
+import {EigenDACertVerifier} from "src/periphery/cert/EigenDACertVerifier.sol";
+
+import {MockStakeRegistry} from "test/mock/MockStakeRegistry.sol";
+import {MockRegistryCoordinator} from "test/mock/MockRegistryCoordinator.sol";
+
+import {InitParamsLib} from "script/deploy/mainnet/DeployMainnetConfig.sol";
+
+import {Script} from "forge-std/Script.sol";
+import {console2} from "forge-std/console2.sol";
+
+contract DeployMainnet is Script {
+    using InitParamsLib for string;
+
+    string constant PROXY_ADMIN = "PROXY_ADMIN";
+    string constant INDEX_REGISTRY = "INDEX_REGISTRY";
+    string constant STAKE_REGISTRY = "STAKE_REGISTRY";
+    string constant SOCKET_REGISTRY = "SOCKET_REGISTRY";
+    string constant BLS_APK_REGISTRY = "BLS_APK_REGISTRY";
+    string constant REGISTRY_COORDINATOR = "REGISTRY_COORDINATOR";
+    string constant THRESHOLD_REGISTRY = "THRESHOLD_REGISTRY";
+    string constant RELAY_REGISTRY = "RELAY_REGISTRY";
+    string constant PAYMENT_VAULT = "PAYMENT_VAULT";
+    string constant DISPERSER_REGISTRY = "DISPERSER_REGISTRY";
+    string constant SERVICE_MANAGER = "SERVICE_MANAGER";
+    string constant EJECTION_MANAGER = "EJECTION_MANAGER";
+    string constant OPERATOR_STATE_RETRIEVER = "OPERATOR_STATE_RETRIEVER";
+    string constant CERT_VERIFIER = "CERT_VERIFIER";
+    string constant PAUSER_REGISTRY = "PAUSER_REGISTRY";
+    string constant EMPTY_CONTRACT = "EMPTY_CONTRACT";
+    string constant MOCK_STAKE_REGISTRY = "MOCK_STAKE_REGISTRY";
+    string constant MOCK_REGISTRY_COORDINATOR = "MOCK_REGISTRY_COORDINATOR";
+
+    mapping(string => address) deployed; // Addresses of the deployed contracts, whether they be a proxy or not.
+    mapping(string => address) impl; // Implementation addresses of the deployed contracts.
+    mapping(string => bool) upgraded; // Whether the deployment of a contract is upgraded to its final implementation. Should beTrue if the contract is not a proxy
+
+    string cfg;
+
+    function initConfig() internal virtual {
+        cfg = vm.readFile(vm.envString("DEPLOY_MAINNET_CONFIG"));
+    }
+
+    function run() public virtual {
+        initConfig();
+        vm.startBroadcast();
+
+        // DEPLOY PROXY ADMIN
+        deployed[PROXY_ADMIN] = address(new ProxyAdmin());
+        impl[PROXY_ADMIN] = deployed[PROXY_ADMIN];
+        upgraded[PROXY_ADMIN] = true;
+
+        // DEPLOY MOCK IMPLEMENTATION
+        impl[EMPTY_CONTRACT] = address(new EmptyContract());
+
+        // DEPLOY PAUSER
+        deployed[PAUSER_REGISTRY] = address(new PauserRegistry(cfg.pausers(), cfg.unpauser()));
```

Comment:
how does cfg have methods? its a string? I'm confused as to how this works.

Reply:
Oh I see you're using `InitParamsLib` on string type.. hmm that feels super hacky to me, but idk if forge allows for typedefs or more sophisticated typing solutions. 

Should prob be documented at least (a docstring on top of the `use initParamsLib` statement would have helped me and future readers here)